src/NoteDataUtil.cpp

```diff
@@ -1935,14 +2073,23 @@ static void SuperShuffleTaps( NoteData &inout, int iStartIndex, int iEndIndex )
 
 void NoteDataUtil::Turn( NoteData &inout, StepsType st, TrackMapping tt, int iStartIndex, int iEndIndex )
 {
-	int iTakeFromTrack[MAX_NOTE_TRACKS];	// New track "t" will take from old track iTakeFromTrack[t]
-	GetTrackMapping( st, tt, inout.GetNumTracks(), iTakeFromTrack );
-
 	NoteData tempNoteData;
-	tempNoteData.LoadTransformed( inout, inout.GetNumTracks(), iTakeFromTrack );
 
-	if( tt == super_shuffle )
-		SuperShuffleTaps( tempNoteData, iStartIndex, iEndIndex );
+	if( tt == hyper_shuffle )
```

Comment:
Why aren't we just adding an else to the `tt == supershuffle` condition?

```cpp
  int iTakeFromTrack[MAX_NOTE_TRACKS];	// New track "t" will take from old track iTakeFromTrack[t]
	GetTrackMapping( st, tt, inout.GetNumTracks(), iTakeFromTrack );

	NoteData tempNoteData;
	tempNoteData.LoadTransformed( inout, inout.GetNumTracks(), iTakeFromTrack );

	if( tt == super_shuffle )
	  SuperShuffleTaps( tempNoteData, iStartIndex, iEndIndex );
	else if( tt == hyper_shuffle )
    HyperShuffleNotes( tempNoteData, iStartIndex, iEndIndex );
    ```

Reply:
SuperShuffle requires holds to be shuffled by a normal shuffle beforehand so it shares code with the other turn mods that do column transformations. HyperShuffle doesn't use the column transform code at all.

Reply:
I see. Makes sense. Can you document this with some comments around?